assets/js/base/components/drawer/style.scss

```diff
@@ -0,0 +1,116 @@
+$drawer-width: 480px;
+
+.wc-block-components-drawer__screen-overlay {
+	background-color: rgba(95, 95, 95, 0.35);
+	bottom: 0;
+	left: 0;
+	position: fixed;
+	right: 0;
+	top: 0;
+	transition: all 0.3s;
```

Comment:
Good catch with the lack of animation on the overlay. I have been playing around with it a a bit more and I noticed it has the same issue as `.wc-block-components-drawer`. When the Drawer is closed, the element doesn't exist in the DOM, so when it's opened it gets appended but then the transition is not triggered. I had to create a `fadein` animation for it to work. Besides that, I copied your suggestion to use opacity.

You can see the changes here: d620a50001cc64bc4f1efa4e117ad16311bea6b1.

> Also, it's not recommended to just transition all properties, because:

Good points, I will keep that in mind in the future!

Reply:
The animation is much smoother now 👏